### DISCUSSION TITLE
v0.9.40-fix3的掉线重连延迟设置失效了？

### DISCUSSION THREAD

**Discussion by mengshouer** [Mrs4s/go-cqhttp#703] [Q&A]
v0.9.40重连延迟还有用，fix1和fix2没试过，fix3是秒重连，设置了延迟设置不生效

**Comment by qianjunakasumi** [comment]
如有可能请提供log

> 帮助我们定位问题：请切换至[v0.9.40-fix1](https://github.com/Mrs4s/go-cqhttp/releases/tag/v0.9.40-fix1)尝试

**Reply by mengshouer** [reply]
![QQ截图20210317122547](https://user-images.githubusercontent.com/36481080/111415164-48a91d80-86d9-11eb-9cc5-8cfadb6a6d66.png)

fix1
![fix1](https://user-images.githubusercontent.com/36481080/111415181-5199ef00-86d9-11eb-880b-4548a8e0f01d.png)

fix4
![fix4](https://user-images.githubusercontent.com/36481080/111415231-65ddec00-86d9-11eb-8f99-4330010dd719.png)

**Reply by Mrs4s** [reply]
有空会排查下, 可能是什么地方改坏了

**Reply by qianjunakasumi** [reply]
我在 #624 中引入的更改

原本设计思想是
考虑到异常断线多为网络环境波动原因
为了高可用
当重连触发时，重连延迟是不发挥作用的，即立即重连恢复连接

只有当连接（重连）失败时 重连延迟设置才会起效

也就是说，
只有尝试连接失效时才会进行延迟重连的流程，

延迟重连功能本身没有失效

关于断线后是立即重连
还是遵循设置的重连延迟
可以讨论下

@Mrs4s
